### PR TITLE
libbpf-tools: fix some minor issues of ext4dist

### PR DESCRIPTION
**Pull request by KentaTada**
Fix below minor issues of ext4dist

* Fix the check of the existence of module BTFs
* Initialize the global variable

Signed-off-by: Kenta Tada <Kenta.Tada@sony.com>

### CODE REVIEW

File: libbpf-tools/ext4dist.c [hunk 1]
```diff
@@ -153,8 +153,8 @@ static bool should_fallback(void)
 	 * to fall back to use kprobe to be compatible with the old kernel.
 	 */
 	if (is_kernel_module("ext4") && !access("/sys/kernel/btf/ext4", R_OK))
-		return true;
-	return false;
+		return false;
+	return true;
```
Comment (anakryiko) [inline libbpf-tools/ext4dist.c:157]: as I mentioned on another PR, I think this is wrong. We fallback to kprobe if ext4 is a module and kernel doesn't support module BTFs. That's what current logic is doing, and I think correctly.

File: libbpf-tools/ext4dist.bpf.c [hunk 2]
```diff
@@ -12,7 +12,7 @@ const volatile pid_t targ_tgid = 0;
 
 #define MAX_ENTRIES	10240
 
-struct hist hists[__MAX_FOP_TYPE];
+struct hist hists[__MAX_FOP_TYPE] = {};
```
Comment (anakryiko) [inline libbpf-tools/ext4dist.bpf.c:15]: this is good, thank you!

**Review by anakryiko — CHANGES_REQUESTED** [review]

File: libbpf-tools/ext4dist.c (hunk 1 above)
Comment (KentaTada) [reply libbpf-tools/ext4dist.c:157]: >as I mentioned on another PR, I think this is wrong. We fallback to kprobe if ext4 is a module and kernel doesn't support module BTFs. That's what current logic is doing, and I think correctly.

I misunderstood the objective of this function. Thanks.

BTW,
If kernel doesn't support module BTFs, I think `access("/sys/kernel/btf/ext4", R_OK)` will not return 0 but return -1.
So, I think this check is ` if (is_kernel_module("ext4") && (access("/sys/kernel/btf/ext4", R_OK) == -1 ))`

Comment (anakryiko) [reply libbpf-tools/ext4dist.c:157]: you are right, `!access()` is wrong here. It should be just `is_kernel_module() && access()` (no need for == -1, enough to check for non-zero return).

Comment (KentaTada) [reply libbpf-tools/ext4dist.c]: Thank for very much for your support.
I fixed it.

**Review by anakryiko — APPROVED** [review]
Looks good, thanks!

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Pull request merged**